.github/workflows/icon-review.yml

```diff
@@ -54,19 +55,29 @@ jobs:
           echo image_url=$IMAGE_URL >> $GITHUB_ENV
         shell: bash
 
-      - name: Post review in PR ✍️
-        uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
+      - name: Find comment if exists 🕵️
+        uses: peter-evans/find-comment@3eae4d37986fb5a8592848f6a574fdf654e61f9e # v3.1.0
+        id: fc
         with:
-          script: |
-            const image_url = "${{ env.image_url }}";
-            console.log('URL: '+image_url);
-            const commentBody = '## Preview \n\nThank you for creating a pull request. This preview shows you how your changes will look on the different themes: \n\n ![Generated Preview](' + image_url + ') \n\n You can find more information how to contribute in the [contribution guidelines](https://github.com/material-extensions/vscode-material-icon-theme/blob/main/CONTRIBUTING.md).';
+          issue-number: ${{ github.event.pull_request.number }}
+          comment-author: 'github-actions[bot]'
+          body-includes: This preview shows you how your changes will look on the different themes
 
-            github.rest.issues.createComment({
-              issue_number: context.issue.number,
-              owner: context.repo.owner,
-              repo: context.repo.repo,
-              body: commentBody
-            })
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Post or update comment in PR ✍️
+        uses: peter-evans/create-or-update-comment@71345be0265236311c031f5c7866368bd1eff043 # v4.0.0
+        with:
+          comment-id: ${{ steps.fc.outputs.comment-id }}
+          issue-number: ${{ github.event.pull_request.number }}
+          body: |
+            ## Preview
+
+            Thank you for creating a pull request. This preview shows you how your changes will look on the different themes:
+
+            <details${{ env.svgFiles < 10 && ' open' || '' }}>
```

Comment:
It is also necessary to implement more complex logic here, possibly using JavaScript

Reply:
How so? The the number of files is less then 10, then be open, otherwise default to closed. It'll be in a details either way if you want to shrink it on a small one. Keeps it consistent.